### PR TITLE
Add TableScanBuilder

### PR DESCRIPTION
**Pull request by majetideepak**
Remove various tableScan APIs.
Add hiveTableScan API to build a Hive Connector TableScanNode. This is sufficient for most use-cases.
Add tpchTableScan API to build a TPC-H Connector TableScanNode.
Add TableScanBuilder API to build a custom TableScanNode.

Resolves: https://github.com/facebookincubator/velox/issues/7391

### CODE REVIEW

**Comment by majetideepak** [comment]
@mbasmanova, this is ready for review.

File: velox/exec/tests/utils/PlanBuilder.h [hunk 1]
```diff
@@ -132,45 +107,139 @@ class PlanBuilder {
...
+    }
+
+    /// @param connectorId The id of the connector to scan.
+    TableScanBuilder& connectorId(const std::string& connectorId) {
+      connectorId_ = connectorId;
+      return *this;
+    }
+
+    /// @param outputType List of column names and types to read from the table.
+    TableScanBuilder& outputType(const RowTypePtr outputType) {
```
Comment (mbasmanova) [inline velox/exec/tests/utils/PlanBuilder.h:152]: drop const + std::move

File: velox/exec/tests/utils/PlanBuilder.h [hunk 2]
```diff
@@ -132,45 +107,139 @@ class PlanBuilder {
...
+      outputType_ = outputType;
+      return *this;
+    }
+
+    /// @param subfieldFilters A list of SQL expressions for the range filters
+    /// to apply to individual columns. Supported filters are: column <= value,
+    /// column < value, column >= value, column > value, column = value, column
+    /// IN (v1, v2,.. vN), column < v1 OR column >= v2.
+    TableScanBuilder& subfieldFilters(
+        const std::vector<std::string>& subfieldFilters) {
```
Comment (mbasmanova) [inline velox/exec/tests/utils/PlanBuilder.h:162]: remove const & + std::move

File: velox/exec/tests/utils/PlanBuilder.h [hunk 3]
```diff
@@ -132,45 +107,139 @@ class PlanBuilder {
...
+      subfieldFilters_ = subfieldFilters;
+      return *this;
+    }
+
+    /// @param remainingFilter SQL expression for the additional conjunct. May
+    /// include multiple columns and SQL functions. The remainingFilter is
+    /// AND'ed with all the subfieldFilters.
+    TableScanBuilder& remainingFilter(const std::string& remainingFilter) {
```
Comment (mbasmanova) [inline velox/exec/tests/utils/PlanBuilder.h:170]: ditto

File: velox/exec/tests/utils/PlanBuilder.h [hunk 4]
```diff
@@ -132,45 +107,139 @@ class PlanBuilder {
...
+
+    /// @param columnAliases Optional aliases for the column names. The key is
+    /// the alias (name in 'outputType'), value is the name in the files.
+    TableScanBuilder& columnAliases(
+        const std::unordered_map<std::string, std::string>& columnAliases) {
+      columnAliases_ = columnAliases;
+      return *this;
+    }
+
+    /// @param tableHandle
```
Comment (mbasmanova) [inline velox/exec/tests/utils/PlanBuilder.h:194]: Redundant comment. Let's remove.

File: velox/exec/tests/utils/PlanBuilder.h [hunk 5]
```diff
@@ -132,45 +107,139 @@ class PlanBuilder {
...
+    TableScanBuilder& tableHandle(
+        const std::shared_ptr<connector::ConnectorTableHandle>& tableHandle) {
+      tableHandle_ = tableHandle;
+      return *this;
+    }
+
+    /// @param assignments
```
Comment (mbasmanova) [inline velox/exec/tests/utils/PlanBuilder.h:201]: broken formatting?

File: velox/exec/tests/utils/PlanBuilder.h [hunk 6]
```diff
@@ -132,45 +107,139 @@ class PlanBuilder {
...
+    }
+
+    /// Stop the TableScanBuilder.
+    PlanBuilder& endTableScan() {
+      planBuilder_.planNode_ = build();
+      return planBuilder_;
+    }
+
+    /// Build the plan node TableScanNode.
+    core::PlanNodePtr build();
```
Comment (mbasmanova) [inline velox/exec/tests/utils/PlanBuilder.h:220]: Should this method be private?

File: velox/exec/tests/utils/PlanBuilder.h [hunk 7]
```diff
@@ -132,45 +107,139 @@ class PlanBuilder {
...
+
+   private:
+    PlanBuilder& planBuilder_;
+    core::PlanNodeId id_;
```
Comment (mbasmanova) [inline velox/exec/tests/utils/PlanBuilder.h:224]: Is this needed? Can it be a local variable in endTableScan?

**Review by mbasmanova — COMMENTED** [review]
@majetideepak Thank you for working on this. This change is very large. Is there a way to break it up into smaller pieces?

**Comment by majetideepak** [comment]
@mbasmanova I opened a smaller PR here https://github.com/facebookincubator/velox/pull/7606 and addressed your comments there.
I will add other changes in subsequent PRs.

**Comment by majetideepak** [comment]
This work has been merged as two other PRs

**Pull request closed without merging**